### ISSUE TITLE
Massive Crash and loss of added boorus to search

### ISSUE THREAD

**Issue by SultrySamthepennanceman** [Bionus/imgbrd-grabber#1081]
### What steps will reproduce the problem?
No idea, I was adding boorus to grabber when an unrelated issue caused a bluescreen crash from my graphics card, and I lost my current Palemoon browsing session, and for whatever reason, my Grabber sources are missing a large bulk of their massive number

### What is the expected behavior? What do you get instead?
When I opened up sources I only had a few of the essentials listed in my sources. I went to C:\Users\USER\AppData\Local\Bionus\Grabber\sites\Gelbooru (0.1) and noticed my added sources were still there, but they're not in the program itself, so I'm wondering if there's any way I can fix this.

### How often does this problem occur? On which sources?
First time this has ever happened to me.

### What version of the program are you using? On what operating system?
5.5.0

### Please provide any additional information below
If you need anything, (Crash logs) then please ask.

**Comment by SultrySamthepennanceman** [comment]
I should note, this probably has some sort of relation to how sources are added to Grabber, so I'm hopeful this can be fixed through editing a file or something.

**Comment by Bionus** [comment]
Are all missing sites from "Gelbooru (0.1)"?
Does the log say anything special when you start the program?

The fastest way to reproduce would be to upload your `sites.txt` file from which sources are missing, maybe there's something inside that could explain the problem.

**Comment by SultrySamthepennanceman** [comment]
Right, I just checked my files in C:\Users\USER\AppData\Local\Bionus\Grabber\sites You're right, this seems to have affected everything in the Gelbooru (0.1) folder. Do you want the sites.txt from that?

**Comment by SultrySamthepennanceman** [comment]
https://anonfile.com/N8A64cc5b4/sites.txt well here's the file I think you want.

**Comment by Bionus** [comment]
It seems to be corrupt beyond repairs, unfortunately. Quite unfortunate that your computer crashed exactly when Grabber was writing inside the file... 😞

**Comment by SultrySamthepennanceman** [comment]
Do I just need to write the names of the sites into the sites.txt file to fix this? the files are still there, and I think it's just the sites.txt that seems to be buggered. Seeing as this can happen during a write, is it possible to have a backup in the future?

**Comment by Bionus** [comment]
> Do I just need to write the names of the sites into the sites.txt file to fix this?

Yes, the `sites.txt` file should contain the domains of all sources, one per line.

> Seeing as this can happen during a write, is it possible to have a backup in the future?

I'll see what I can do to automatically use a backup feature when writing these files (`sites.txt`, `tabs.txt`, etc.).

**Comment by SultrySamthepennanceman** [comment]
Hm, I've added some of my Gelbooru sources back to the sites.text and now I get crashes when opening up the sources menu in the program.
